src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java

```diff
@@ -40,4 +40,11 @@ public boolean deleteById(Long userId) {
     public List<User> getAll() {
         return Storage.users;
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(user -> user.getLogin().equals(login))
```

Comment:
comment about equalsignorecase should be here))

Reply:
I would choose to use equals()) but that'll be good if you will find some articles about any of these approach. WDYT, @tarkaiev ?